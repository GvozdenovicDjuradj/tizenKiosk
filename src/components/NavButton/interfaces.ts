import { ViewStyle } from "react-native"
import { ButtonProps } from "../Button"

export interface Styles {
  navButton: ViewStyle
  navButtonLeft: ViewStyle
  navButtonRight: ViewStyle
  triangleLeft: ViewStyle
  triangleRight: ViewStyle
  view: ViewStyle
}

export interface NavButtonProps extends ButtonProps {
  buttonColor?: string
  privacyPolicyButtonDefaultText?: string
}

export interface NavButtonNextState {
  isDisabled: boolean
}
