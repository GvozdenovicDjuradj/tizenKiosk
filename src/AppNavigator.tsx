import React from "react"
import { BackHandler } from "react-native"
import { connect } from "react-redux"
import { createStackNavigator, NavigationInjectedProps } from "react-navigation"
import { AppScreens, RootState } from "./interfaces"
import { selectors } from "./utils"
import ConfirmationScreen from "./components/confirmationScreen"
import CustomerScreen from "./components/customerScreen"
import Home from "./components/Home"
import ServiceSelectionScreen from "./components/serviceSelection"
import SubserviceSelectionScreen from "./components/serviceSelection/SubServiceSelection"
import ServiceInfoScreen from "./components/serviceSelection/ServiceInfoScreen"
import NoConnectionModal from "./components/NoConnectionModal"
import CheckInScreen from "./components/checkInScreen"
import CheckInConfirmation from "./components/checkInConfirmationScreen"
import EventCheckInScreen from "./components/eventCheckInScreen"
import EventCheckInConfirmationScreen from "./components/eventCheckInConfirmationScreen"
import KioskClosedScreen from "./components/kioskClosedScreen"
import QuestionsScreen from "./components/QuestionsScreen"
import QueueFullOrNAScreen from "./components/QueueFullOrNAScreen"
import ScreenSaver from "./components/screenSaver"
import { handleBackButton, removeBackButtonHandler } from "./utils/backButton"
import PrivacyPolicyScreen from "./components/privacyPolicyScreen"
import KeepAwake from 'react-native-keep-awake'
import QueueUnderCapacityScreen from "./components/QueueUnderCapacityScreen"

const AppStackNavigator = createStackNavigator({
  [AppScreens.HOME]: { screen: Home },
  [AppScreens.SERVICE_SELECTION]: { screen: ServiceSelectionScreen },
  [AppScreens.SUBSERVICE_SELECTION]: { screen: SubserviceSelectionScreen },
  [AppScreens.SERVICE_INFO]: { screen: ServiceInfoScreen },
  [AppScreens.CUSTOMER_DETAILS]: { screen: CustomerScreen },
  [AppScreens.SCREEN_SAVER]: { screen: ScreenSaver },
  [AppScreens.QUEUE_CONFIRMATION]: {
    screen: ConfirmationScreen,
    navigationOptions: {
      gesturesEnabled: false,
    },
  },
  [AppScreens.QUEUE_FULL_OR_NA]: { screen: QueueFullOrNAScreen },
  [AppScreens.QUEUE_UNDER_OCCUPANCY]: { screen: QueueUnderCapacityScreen },
  [AppScreens.QUESTIONS]: { screen: QuestionsScreen },
  [AppScreens.CHECK_IN]: { screen: CheckInScreen },
  [AppScreens.EVENT_CHECK_IN]: { screen: EventCheckInScreen },
  [AppScreens.EVENT_CHECK_IN_CONFIRMATION]: { screen: EventCheckInConfirmationScreen },
  [AppScreens.NO_CONNECTION_MODAL]: { screen: NoConnectionModal },
  [AppScreens.KIOSK_CLOSED]: { screen: KioskClosedScreen },
  [AppScreens.CHECK_IN_CONFIRMATION]: {
    screen: CheckInConfirmation,
    navigationOptions: {
      gesturesEnabled: false,
    },
  },
  [AppScreens.PRIVACY_POLICY]: { screen: PrivacyPolicyScreen },
}, {
  cardStyle: { backgroundColor: "transparent" },
  headerMode: "none",
  initialRouteName: AppScreens.HOME,
  mode: "card",
})

interface Props extends NavigationInjectedProps {
  loading: boolean,
  hasKioskModeEnable: boolean
}

class AppNavigator extends React.Component<Props> {
  public static router = AppStackNavigator.router

  /**
   * Depending on hasKioskModeEnable props, calling back button will Quit the app or do nothing
   * react navigation back button handler will overwrite that behavior
   */
  public handleBackButtonClick = () => {
   if (!this.props.hasKioskModeEnable) {
     BackHandler.exitApp()
   }
  }

  public componentWillMount() {
    handleBackButton(this.handleBackButtonClick)
  }

  public componentWillUnmount() {
    removeBackButtonHandler(this.handleBackButtonClick)
  }

  public render() {
    const { hasKioskModeEnable } = this.props
    return (
      <React.Fragment>
        { hasKioskModeEnable &&
          <KeepAwake />
        }
        <AppStackNavigator navigation={this.props.navigation} />
      </React.Fragment>
    )
  }
}

const mapStateToProps = (state: RootState) => ({
  hasKioskModeEnable: selectors.isKioskModeEnabled(state)
})

export default connect(mapStateToProps)(AppNavigator)
