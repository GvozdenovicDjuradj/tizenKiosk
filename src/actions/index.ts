export * from "./app"
export * from "./error"
export * from "./kiosk"
export * from "./kioskSettings"
export * from "./printing"
export * from "./navigationActions"
export * from "./screenSaver"
