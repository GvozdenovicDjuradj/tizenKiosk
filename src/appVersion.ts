import { Platform } from "react-native"
import DeviceInfo from "react-native-device-info"

export default Platform.select({
  web: "n/a",
  default: DeviceInfo.getReadableVersion()
})
