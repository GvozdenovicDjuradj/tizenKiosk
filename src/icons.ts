export default {
    checkboxOff: require("../assets/icons/off.png"),
    checkboxOn: require("../assets/icons/on.png"),
    exclamation: require("../assets/icons/exclamation.png"),
    paper: require("../assets/icons/paper.png"),
    plug: require("../assets/icons/plug.png"),
}
