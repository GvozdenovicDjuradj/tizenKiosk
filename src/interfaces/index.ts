export * from "./answers"
export * from "./app"
export * from "./appState"
export * from "./checkIn"
export * from "./customerInQueue"
export * from "./eventCheckIn"
export * from "./kiosk"
export * from "./modal"
export * from "./navigation"
export * from "./product"
export * from "./question"
export * from "./routes"
export * from "./utils"
export * from "./validation"
export * from "./venue"
export * from "./screenSaver"

import { AppState } from "./app"
import { CheckInState } from "./checkIn"
import { EventCheckInState } from "./eventCheckIn"
import { KioskSettingsForm } from "../reducers/kioskSettings"
import { KioskState } from "./kiosk"
import { ModalState } from "./modal"
import { QuestionAnswer } from "./answers"
import { QuestionsReducer } from "./question"
import { ValidationState } from "./validation"
import { PrintingState } from "./printing"
import { PrinterState } from "../printer/reducer"
import { ScreenSaverState } from "./screenSaver"

export interface Flags {
  [key: string]: any
}

export type ValidatorFn = (text: string, ...args: any[]) => { error: string }

export interface RootState {
  answers: QuestionAnswer[];
  app: AppState;
  checkIn: CheckInState;
  error: string;
  eventCheckIn: EventCheckInState;
  kiosk: KioskState;
  kioskSettings: KioskSettingsForm;
  modal: ModalState;
  printer: PrinterState;
  printing: PrintingState;
  questions: QuestionsReducer;
  validation: ValidationState;
  screenSaver?: ScreenSaverState;
}
