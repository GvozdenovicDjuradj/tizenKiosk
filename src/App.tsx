import React, { Component } from "react"
import { Store } from "redux"
import { Provider } from "react-redux"
import { StatusBar, View, PermissionsAndroid, Platform } from "react-native"
import { PersistGate } from "redux-persist/integration/react"
import { Navigator as NavigatorType, setNavigator } from "./navigation"
import Navigator from "./routes"
import { applicationStart, initApp as initAppActionCreator, stopApp } from "./actions/app"
import { RootState } from "./interfaces"
import { persistor, store } from "./store"
import { setup } from "./utils/localeString"
import Modals from "./modals"
import Printing from "./components/Printing"
import Ticket from "./printer/containers/Ticket"
import Notification from "./components/Notification"
import initSentry from "./initSentry"
import { getKioskIdAndDomainFromUrl } from "./actions"
import FadeNotification from "./printer/containers/FadeNotification"

if (Platform.OS !== "web") {
    initSentry()
}

setup(store)

if (Platform.OS === "web") {
    setTimeout(() => {
        store.dispatch(getKioskIdAndDomainFromUrl())
    }, 0)
}

setTimeout(() => store.dispatch(applicationStart()))

const start = (store$: Store<RootState>, navigator: NavigatorType) => {
    if (navigator) {
        setNavigator(navigator)
        store$.dispatch(initAppActionCreator())

    } else {
        store$.dispatch(stopApp())
    }
}

export default class App extends Component {
    public async componentDidMount() {
        if (Platform.OS === "android") {
            // TODO - This should be re-prompted in settings later, I will handle this in a later ticket
            await PermissionsAndroid.request(
              PermissionsAndroid.PERMISSIONS.READ_EXTERNAL_STORAGE,
              {
                title: "Qudini Kiosk - Printing",
                message: `We require access to read a configuration
                file which stores internal settings for the connected
                printer, in addition we also need to read saved tickets
                which we save to external storage as an image.`,
                buttonNegative: "Decline",
                buttonPositive: "Accept",
              },
            )

            await PermissionsAndroid.request(
              PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE,
              {
                title: "Qudini Kiosk - Printing",
                message: `We require access to write to a configuration
                  file which stores internal settings for the connected
                  printer, in addition we also need to write saved tickets
                  which we save to external storage as an image.`,
                buttonNegative: "Decline",
                buttonPositive: "Accept",
              },
            )
        }
    }

    public render() {
        return (
            <Provider store={store}>
                <PersistGate persistor={persistor}>
                    <View style={{ flex: 1 }}>
                        <StatusBar hidden/>
                        <Ticket/>
                        <View style={{ flex: 1, width: "100%", height: "100%", overflow: "hidden" }}>
                            <Modals/>
                            <Notification/>
                            <Navigator ref={(navigator: NavigatorType) => start(store, navigator)}/>
                            <Printing />
                        </View>
                        <FadeNotification />
                    </View>
                </PersistGate>
            </Provider>
        )
    }
}
