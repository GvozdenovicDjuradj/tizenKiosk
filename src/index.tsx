import "babel-polyfill"
import { AppRegistry } from "react-native"
import App from "./App"
import app from "../app.json"

AppRegistry.registerComponent(app.name, () => App)

if (window && typeof window === "object" && window.document) {
  AppRegistry.runApplication(app.name, { rootTag: document.getElementById("app") })
}
