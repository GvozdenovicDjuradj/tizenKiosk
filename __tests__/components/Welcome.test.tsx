import React from "react"
// import { Image, TouchableOpacity } from "react-native"
import { Provider } from "react-redux"
import renderer from "react-test-renderer"
import mockStore from "redux-mock-store"
import Welcome from "../../src/components/welcome"
// import Content from "../../src/components/welcome/Content"
// import LanguageSelector from "../../src/components/LanguageSelector"
// import CircleButton from "../../src/components/CircleButton"
import { AppScreens, KioskTemplate, Venue, PrivacyPolicy } from "../../src/interfaces"
import { createTemplate, createPrivacyPolicy } from "../fixtures"

jest.mock("../../src/utils/getRouteName", () => ({
  getRouteName: () => "HOME"
}))

describe("Welcome component tests", () => {
  let template: KioskTemplate
  let venue: Partial<Venue>
  let privacyPolicy: PrivacyPolicy
  const languages = {
    mainLanguage: {
      countryCallingCode: "+44",
      countryIsoCode: "GB",
      countryName: "United Kingdom",
      languageIsoCode: "en",
      languageName: "en"
    },
    otherLanguages: [{
      countryCallingCode: "+380",
      countryIsoCode: "UA",
      countryName: "Ukraine",
      languageIsoCode: "ua",
      languageName: "ua"
    }],
    translations: {},
  }

  beforeEach(() => {
    template = createTemplate()
    template.languages = languages
    venue = {
      merchant: {
        featureSettings: {
          hasQudiniBrand: true
        },
        id: 123
      }
    }
    privacyPolicy = createPrivacyPolicy()
  })

  afterAll(() => {
    jest.unmock("../../src/utils/getRouteName")
  })

  it.skip("Should render Welcome component successfully", () => {
    const store = mockStore()({
      app: { initialScreen: undefined },
      kiosk: { fields: { url: "https://qa.qudini.com" } },
    })
    const tree = renderer.create(
      <Provider store={store}>
        <Welcome />
      </Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("Should render Welcome component with backgound image and language selector", () => {
    const store = mockStore()({
      app: { initialScreen: AppScreens.HOME },
      kiosk: {
        fields: { url: "https://qa.qudini.com" },
        settings: { template, venue },
        privacyPolicy,
      },
    })
    const tree = renderer.create(
      <Provider store={store}>
        <Welcome />
      </Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  // it("Should dispatch secret tap action", () => {
  //   const store = mockStore()({
  //     app: { initialScreen: undefined },
  //     kiosk: { fields: { url: "https://qa.qudini.com" } },
  //   })
  //   const wrapper = shallow(
  //     <Welcome />,
  //     { context: { store } }
  //   )
  //   expect(wrapper.dive().find(TouchableOpacity).length).toBe(1)
  //   wrapper.dive().find(TouchableOpacity).simulate("press")
  //   expect(store.getActions()).toMatchSnapshot()
  // })

  // it(`Should dispatch "Go to service selection" navigation action`, () => {
  //   const store = mockStore()({
  //     app: { initialScreen: undefined },
  //     kiosk: {
  //       fields: { url: "https://qa.qudini.com" },
  //       settings: { template, venue },
  //     },
  //   })
  //   const wrapper = shallow(
  //     <Welcome />,
  //     { context: { store } }
  //   )
  //   expect(wrapper.dive().find(Content).length).toBe(1)
  //   const contentWrapper = wrapper.dive().find(Content)
  //   expect(contentWrapper.dive().find(CircleButton).length).toBe(1)
  //   contentWrapper.dive().find(CircleButton).at(0).simulate("press")
  //   expect(store.getActions()).toMatchSnapshot()
  // })

  // it(`Should dispatch "Go to check-in" navigation action`, () => {
  //   template.enableBookingCheckin = true
  //   const store = mockStore()({
  //     app: { initialScreen: undefined },
  //     kiosk: {
  //       fields: { url: "https://qa.qudini.com" },
  //       settings: { template, venue },
  //     },
  //   })
  //   const wrapper = shallow(
  //     <Welcome />,
  //     { context: { store } }
  //   )
  //   expect(wrapper.dive().find(Content).length).toBe(1)
  //   const contentWrapper = wrapper.dive().find(Content)
  //   expect(contentWrapper.dive().find(CircleButton).length).toBe(2)
  //   contentWrapper.dive().find(CircleButton).at(1).simulate("press")
  //   expect(store.getActions()).toMatchSnapshot()
  // })

  // it(`Should dispatch "Go to event check-in" navigation action`, () => {
  //   template.enableBookingCheckin = true
  //   template.enableEventCheckin = true
  //   const store = mockStore()({
  //     app: { initialScreen: undefined },
  //     kiosk: {
  //       fields: { url: "https://qa.qudini.com" },
  //       settings: { template, venue },
  //     },
  //   })
  //   const wrapper = shallow(
  //     <Welcome />,
  //     { context: { store } }
  //   )
  //   expect(wrapper.dive().find(Content).length).toBe(1)
  //   const contentWrapper = wrapper.dive().find(Content)
  //   expect(contentWrapper.dive().find(CircleButton).length).toBe(3)
  //   contentWrapper.dive().find(CircleButton).at(2).simulate("press")
  //   expect(store.getActions()).toMatchSnapshot()
  // })

})
