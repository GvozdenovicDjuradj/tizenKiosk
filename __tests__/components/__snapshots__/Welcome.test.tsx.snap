// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome component tests Should render Welcome component with backgound image and language selector 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "height": 100,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": "100%",
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "width": 80,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#F80347",
          "justifyContent": "center",
          "marginTop": 80,
          "padding": 20,
          "width": "100%",
        },
        Object {
          "backgroundColor": "#00ff00",
        },
        Object {
          "marginTop": 0,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Arial",
            "fontSize": 55,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          Object {
            "color": "#fff",
            "fontFamily": "Arial",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flex": 3,
        "paddingHorizontal": 50,
        "paddingTop": 40,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00ff00",
            "borderRadius": 130,
            "elevation": 7,
            "height": 260,
            "justifyContent": "center",
            "marginBottom": 10,
            "opacity": 1,
            "padding": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 10,
            "width": 260,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "bottom": 15,
        "justifyContent": "center",
        "position": "absolute",
        "right": 50,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "printer.png",
        }
      }
      style={
        Object {
          "height": 50,
          "resizeMode": "contain",
          "width": 170,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "flexDirection": "row",
          "height": 80,
          "justifyContent": "flex-start",
          "paddingHorizontal": 20,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 10,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://qa.qudini.com/static/gb.png",
            }
          }
          style={
            Object {
              "height": 45,
              "width": 60,
            }
          }
        />
        <Text>
          en
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 10,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://qa.qudini.com/static/ua.png",
            }
          }
          style={
            Object {
              "height": 45,
              "width": 60,
            }
          }
        />
        <Text>
          ua
        </Text>
      </View>
    </View>
  </View>
</View>
`;
